src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -384,15 +401,43 @@ private static Vector2 PatchCenter(Vector2 pos)
             pos.Y + Constants.PATCH_NODE_RECT_LENGTH * 0.5f);
     }
 
-    private Line2D CreateConnectionLine(Vector2[] points, Color connectionColor)
+    private static Vector2 ClosestPatchSideCenter(Vector2 patchPos, Vector2 pointPos, bool isVertical)
+    {
+        float halfLength = Constants.PATCH_NODE_RECT_LENGTH * 0.5f;
+        float posX = patchPos.X;
+        float posY = patchPos.Y;
+        List<Vector2> sideCenterPoints = new();
```

Comment:
This method should be rewritten to not use this temporary memory. As we are working on a realtime game, memory allocations should be avoided whenever possible. Here it looks like it will be possible by writing more code to manually compare the values to get the actual single result value.